rustual-boy-serialization/src/version1.rs

```diff
@@ -1,4 +1,39 @@
 use std::collections::HashSet;
+use serde_bytes;
+
+mod serde_ibytes {
```

Comment:
This seems like a pretty generic module; arguably more generic than the `version1` module, so maybe this should be moved into its own file. I suspect we'll end up using this for future versioned state structs too.